### PR TITLE
Move-Audit-2

### PR DESCRIPTION
**Pull request by simsekgokhan**
Audit of latest move changes after 7273897c9db5889c5638390072eff52bc81a8615

### CODE REVIEW

File: language/diem-framework/modules/0L/Subsidy.move [hunk 1]
```diff
@@ -145,29 +144,30 @@ address 0x1 {
...
       assert(vm_addr == CoreAddresses::DIEM_ROOT_ADDRESS(), Errors::requires_role(190104));
 
       // Get eligible validators list
       let genesis_validators = ValidatorUniverse::get_eligible_validators(vm_sig);
       let len = Vector::length(&genesis_validators);
-      let subsidy = 11000000; // ten coins for validator, sufficient for first epoch of transactions, and an extra which the validator will send to operator.
-
+      // ten coins for validator, sufficient for first epoch of transactions,
+      // and an extra which the validator will send to operator.
+      let subsidy = 11000000; // todo: 10 or 11? comment says different
```
Comment (simsekgokhan) [inline language/diem-framework/modules/0L/Subsidy.move:168]: @0o-de-lally FYI.

File: language/diem-framework/modules/0L/Testnet.move [hunk 2]
```diff
@@ -13,8 +13,12 @@ module Testnet {
 
     struct IsTestnet has key { }
 
+    // Todo: never used, is this ok?
```
Comment (simsekgokhan) [inline language/diem-framework/modules/0L/Testnet.move:16]: @0o-de-lally FYI.

File: language/diem-framework/modules/0L/Testnet.move [hunk 3]
```diff
@@ -42,13 +49,18 @@ module StagingNet {
...
+            Errors::requires_role(190301)
+        );
         move_to(account, IsStagingNet{})
     }
 
     public fun is_staging_net(): bool {
         exists<IsStagingNet>(CoreAddresses::DIEM_ROOT_ADDRESS())
     }
 
+    // Todo: no remove_staging_net()? Can this be a problem in production?
```
Comment (simsekgokhan) [inline language/diem-framework/modules/0L/Testnet.move:63]: @0o-de-lally FYI.

File: language/diem-framework/modules/0L/MinerState.move [hunk 4]
```diff
@@ -35,38 +36,39 @@ address 0x1 {
...
 
-  public fun epoch_reset(vm: &signer) acquires MinerStats {
-    CoreAddresses::assert_vm(vm);
-    let state = borrow_global_mut<MinerStats>(CoreAddresses::VM_RESERVED_ADDRESS());
-    state.proofs_in_epoch = 0;
-    state.validator_proofs = 0;
-    state.fullnode_proofs = 0;
-   }
+    // Note: Used only in tests
+    public fun epoch_reset(vm: &signer) acquires MinerStats {
```
Comment (simsekgokhan) [inline language/diem-framework/modules/0L/MinerState.move:55]: @0o-de-lally Should this fn be a test_helper and `assert testnet` added?

**Comment by simsekgokhan** [comment]
@0o-de-lally Ready for review but we might do changes for my questions/comments above.

File: language/diem-framework/modules/0L/Testnet.move (hunk 2 above)
Comment (0o-de-lally) [reply language/diem-framework/modules/0L/Testnet.move:16]: I think this is called externally, for example by test harnesses

File: language/diem-framework/modules/0L/Testnet.move (hunk 3 above)
Comment (0o-de-lally) [reply language/diem-framework/modules/0L/Testnet.move:63]: StagingNet is only used for tests, it's not ever launched in prod.

**Comment by simsekgokhan** [comment]
No further changes planned, ready for review/merge.

**Review by 0o-de-lally — APPROVED** [review]
LGTM

**Pull request merged**